react/package.json

```diff
@@ -41,11 +43,56 @@
     "format": "prettier --write src"
   },
   "eslintConfig": {
-    "extends": "react-app"
+    "extends": "react-app",
+    "plugins": [
+      "import",
+      "mui-unused-classes"
+    ],
+    "rules": {
+      "import/no-useless-path-segments": "warn",
+      "import/no-cycle": "warn",
+      "import/order": [
+        "warn",
+        {
+          "groups": [
+            "builtin",
+            "external",
+            [
+              "parent",
+              "sibling",
+              "index"
+            ]
+          ],
+          "pathGroups": [
+            {
+              "pattern": "react",
+              "group": "external",
+              "position": "before"
+            }
+          ],
+          "pathGroupsExcludedImportTypes": [
+            "react"
+          ],
+          "alphabetize": {
+            "order": "asc",
+            "caseInsensitive": true
+          }
+        }
+      ],
+      "mui-unused-classes/unused-classes": "warn",
+      "react/jsx-curly-brace-presence": "warn",
+      "react/jsx-key": "warn",
+      "sort-imports": [
+        "warn",
+        {
+          "ignoreDeclarationSort": true,
+          "ignoreCase": true
+        }
+      ]
+    }
   },
   "prettier": {
     "printWidth": 100,
-    "tabWidth": 4,
```

Comment:
Is this the only way to make it work?

Reply:
Nope -- a separate formatter could be set for json and prettier wouldn't see the json files anymore. But if editorconfig is handling the tab settings for all files already, prettier will obey them as long as there isn't an explicit override. Removing this setting tells prettier to do what editorconfig says, which means it can format json too.